src/Parser.js

```diff
@@ -488,12 +474,14 @@ Parser.prototype.parseImplicitGroup = function(pos) {
  *
  * @return {?ParseResult}
  */
-Parser.prototype.parseFunction = function(pos) {
-    var baseGroup = this.parseGroup(pos);
+Parser.prototype.parseFunction = function(baseGroup) {
+    if (!baseGroup) {
+        baseGroup = this.parseGroup();
+    }
```

Comment:
It feels a little weird having this as an optional parameter, but I don't have a good suggestion of how to make it less weird.  Could you update the jsdocs to reflect that `baseGroup` is optional?